### PR TITLE
Provide performance workers for C#

### PR DESCRIPTION
**Pull request by jtattermusch**
-- add Grpc.IntegrationTesting.QpsWorker.exe that works with C++ qps_driver
-- implement simple client runner (supports sync unary with closed loop)
-- implement simple server runner (supports async server).

### CODE REVIEW

File: src/csharp/Grpc.IntegrationTesting.QpsWorker/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Reflection;
```
Comment (vjpai) [inline src/csharp/Grpc.IntegrationTesting.QpsWorker/Properties/AssemblyInfo.cs]: Does this file need a copyright comment also? I assume that it does.

Comment (jtattermusch) [inline src/csharp/Grpc.IntegrationTesting.QpsWorker/Properties/AssemblyInfo.cs]: Turns out any of my AssemblyInfo.cs files doesn't contain the copyright comment so I'd prefer to add them in a separate pull requests. Also AssemblyInfo.cs doesn't contain any actual code, so I think it's fine to address this later.

File: src/csharp/Grpc.IntegrationTesting/BenchmarkServiceImpl.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        }
+
+        public async Task StreamingCall(IAsyncStreamReader<SimpleRequest> requestStream, IServerStreamWriter<SimpleResponse> responseStream, ServerCallContext context)
+        {
+            await requestStream.ForEachAsync(async request =>
+            {
+                var response = new SimpleResponse { Payload = CreateZerosPayload(responseSize) };
+                await responseStream.WriteAsync(response);
+            });
+        }
```
Comment (vjpai) [inline src/csharp/Grpc.IntegrationTesting/BenchmarkServiceImpl.cs]: Sorry, I can get most of this C++ but don't exactly understand the above. Can you pass the above function to another reviewer?

File: src/csharp/Grpc.IntegrationTesting/ClientRunners.cs [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            };
+            var stopwatch = new Stopwatch();
+
+            while (!stoppedCts.Token.IsCancellationRequested)
+            {
+                stopwatch.Restart();
+                client.UnaryCall(request);
+                stopwatch.Stop();
+
+                // TODO: 1e9 needs to be in sync with C++ code
```
Comment (vjpai) [inline src/csharp/Grpc.IntegrationTesting/ClientRunners.cs]: Can you file an issue about the histogram and stats? We should track the configurability of that.

File: src/csharp/Grpc.IntegrationTesting/Payloads.cs [hunk 4]
```diff
@@ -0,0 +1,580 @@
+// Generated by the protocol buffer compiler.  DO NOT EDIT!
```
Comment (vjpai) [inline src/csharp/Grpc.IntegrationTesting/Payloads.cs]: As this file is generated code, is it supposed to be included in the repo? Does it need a review?

File: src/csharp/Grpc.IntegrationTesting/WorkerServiceImpl.cs [hunk 5]
```diff
@@ -0,0 +1,96 @@
+#region Copyright notice and license
+
```
Comment (vjpai) [inline src/csharp/Grpc.IntegrationTesting/WorkerServiceImpl.cs]: I would request another reviewer like @murgatroid99 for this file as well.

**Comment by vjpai** [comment]
I've given my comments wherever possible but prefer another file for the places where C++ -> C# conversion is less clear to me. So I woud LGTM except for those 2 files in particular. Just as a side note, do we have a C# style guide or best practices guide that needs following?

File: src/csharp/Grpc.IntegrationTesting/Payloads.cs (hunk 4 above)
Comment (jtattermusch) [inline src/csharp/Grpc.IntegrationTesting/Payloads.cs]: For now, generated C# code gets checked into the repo. Ideally, we wouldn't do this, but it is not so simple to make this work with visual studio projects in a way that wouldn't lead to some visual studio awkwardness. So checking in generated code was the intention here.

**Comment by jtattermusch** [comment]
@murgatroid99  can you please review WorkerServiceImpl.cs and BenchmarkServiceImpl.cs as requested by Vijay?

**Comment by jtattermusch** [comment]
regarding the C# style:   there is no official google-wide C# style as of now. gRPC C# is using stylecop tool to check some basic style policies (I usually run those checks manually and fix what needs to be fixed).

**Pull request merged**